3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py

```diff
@@ -63,46 +64,91 @@
 
 all_GSEA_results = []
 all_signatures = []
-#negative_control = []
-for col in signature.iloc[:,1:50].columns:
+neg_GSEA_results = []
+negative_control = []
+
+for col in signature.iloc[:,1:101].columns:
     df = signature.iloc[:,[0,int(col)]]
     result = blitz.gsea(df, library)
     all_GSEA_results.append(result.assign(z_dim=f"z_{col}"))
     all_signatures.append(df)
-    #add line here to create a negative control--this will be a random scramble of the gene scores
+    neg_df = df.copy()
+    neg_df[col] = neg_df[col].sample(frac=1).reset_index(drop=True)
+    neg_result = blitz.gsea(neg_df, library)
+    neg_GSEA_results.append(neg_result.assign(z_dim=f"z_{col}"))
+    negative_control.append(neg_df)
+
 all_GSEA_results
+neg_GSEA_results
```

Comment:
why print this out?

Reply:
(same can be said of `all_GSEA_results`!)

Reply:
I put those there a while ago to see what each output looked like, but will be deleting before merge